mtags/src/main/scala/scala/meta/internal/mtags/OnDemandSymbolIndex.scala

```diff
@@ -130,17 +127,55 @@ final class OnDemandSymbolIndex(
   private def findSymbolDefinition(
       querySymbol: Symbol
   ): List[SymbolDefinition] = {
-    dialectBuckets.values.toList.flatMap(_.query(querySymbol))
+    dialectBuckets.values
+      .flatMap(_.query(querySymbol))
+      // prioritize defs where found symbols is exact
+      .sortBy(d => !d.isExact)
   }
 
 }
 
 object OnDemandSymbolIndex {
+
   def empty(
       onError: PartialFunction[Throwable, Unit] = { case NonFatal(e) =>
         throw e
       },
       toIndexSource: AbsolutePath => AbsolutePath = identity
-  ): OnDemandSymbolIndex =
-    new OnDemandSymbolIndex(TrieMap.empty, onError, toIndexSource)
+  ): OnDemandSymbolIndex = {
+    new OnDemandSymbolIndex(new Buckets, onError, toIndexSource)
+  }
+
+  class Buckets {
+
+    private val ref =
+      new AtomicReference[List[(Dialect, SymbolIndexBucket)]](List.empty)
+
+    def getOrCreateBucket(
+        dialect: Dialect,
+        mtags: Mtags,
+        toIndexSource: AbsolutePath => AbsolutePath
+    ): SymbolIndexBucket = {
+      val curr = ref.get
+      curr.find(_._1 == dialect) match {
+        case Some((_, bucket)) => bucket
+        case None =>
+          val bucket = SymbolIndexBucket.empty(dialect, mtags, toIndexSource)
+          val entry = (dialect, bucket)
+          val next = curr match {
+            // keep scala3 bucket at first position
+            // it's required for getting proper definitions for sources from `stdLibPatches`
+            case (h @ (firstDialect, _)) :: tl
```

Comment:
thx for noticing! Reverted this this thing.
I initially wanted to avoid sorting per each request but anyway added at after and forgot that then this thing doesn't makes sense

Reply:
Thanks! LGTM once more!